log/testutil.go

```diff
@@ -0,0 +1,122 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+// FieldValidator facilitates testing of Span.Log*() implementations.
+//
+// FieldValidator SHOULD ONLY BE USED FOR TESTS. It is not intended for
+// production code.
+//
+// Usage:
+//
+//     fv := log.NewFieldValidator(t, someLogStructure.Fields)
+//     fv.
+//         ExpectNextFieldEquals("key1", reflect.String, "some string value").
+//         ExpectNextFieldEquals("key2", reflect.Uint32, "4294967295")
```

Comment:
for instance, [this is the only example](https://github.com/opentracing/basictracer-go/pull/37/files#diff-712c45fed6eb613b9582291e641be233R40) of actually using Field values in basictracer. A real recorder would have to call the visitor at this point and convert the fields into some other structure where it has access to private fields, either its own struct or serialization into some thrift/protobuf, from where the tests can read the values.